js_modules/dagster-ui/packages/ui-core/src/launchpad/useLaunchMultipleRunsWithTelemetry.oss.tsx

```diff
@@ -0,0 +1,62 @@
+import {useCallback} from 'react';
+import {useHistory} from 'react-router-dom';
+
+import {useMutation} from '../apollo-client';
+import {TelemetryAction, useTelemetryAction} from '../app/Telemetry';
+import {showLaunchError} from '../launchpad/showLaunchError';
+import {
+  LAUNCH_MULTIPLE_RUNS_MUTATION,
+  LaunchBehavior,
+  handleLaunchMultipleResult,
+} from '../runs/RunUtils';
+import {
+  LaunchMultipleRunsMutation,
+  LaunchMultipleRunsMutationVariables,
+} from '../runs/types/RunUtils.types';
+
+export function useLaunchMultipleRunsWithTelemetry() {
+  const [launchMultipleRuns] = useMutation<
+    LaunchMultipleRunsMutation,
+    LaunchMultipleRunsMutationVariables
+  >(LAUNCH_MULTIPLE_RUNS_MUTATION);
+
+  const logTelemetry = useTelemetryAction();
+  const history = useHistory();
+
+  return useCallback(
+    async (variables: LaunchMultipleRunsMutationVariables, behavior: LaunchBehavior) => {
+      const jobNames = variables.executionParamsList.map((params) => params.selector?.jobName);
+
+      if (
+        jobNames.length !== variables.executionParamsList.length ||
+        jobNames.includes(undefined)
```

Comment:
why would this include `undefined`?

Reply:
Since jobName here is a potentially undefined from the GraphQL type, I decided to add a check here. Not sure in what case it would happen though. Do you want me to remove that check?

https://github.com/dagster-io/dagster/blob/master/js_modules/dagster-ui/packages/ui-core/src/graphql/types.ts#L2148